parser-typechecker/src/Unison/Syntax/TermParser.hs

```diff
@@ -1106,66 +1113,69 @@ substImports ns imports =
           NamesWithHistory.hasTypeNamed (Name.unsafeFromVar full) ns
       ]
 
-block' :: Var v => IsTop -> String -> P v (L.Token ()) -> P v b -> TermP v
+block' :: Var v => IsTop -> String -> P v (L.Token ()) -> P v (L.Token ()) -> TermP v
 block' isTop = block'' isTop False
 
 block'' ::
-  forall v b.
+  forall v.
   Var v =>
   IsTop ->
   Bool -> -- `True` means insert `()` at end of block if it ends with a statement
   String ->
   P v (L.Token ()) ->
-  P v b ->
+  P v (L.Token ()) ->
   TermP v
 block'' isTop implicitUnitAtEnd s openBlock closeBlock = do
   open <- openBlock
   (names, imports) <- imports
   _ <- optional semi
   statements <- local (\e -> e {names = names}) $ sepBy semi statement
-  _ <- closeBlock
-  substImports names imports <$> go open statements
+  _close <- closeBlock
+  theBlock <- substImports names imports <$> go open statements
+  pure $ ABT.annotate (ann open <> ann theBlock) theBlock
```

Comment:
Add `open` to the annotation so that the `do` or `let` or w/e is included in the let-rec annotation.

Reply:
Is this going to include the = or -> as part of the annotation? We don't want that. For elements like that, the block annotation should begin with the first statement in the block.

Reply:
It would yes, I think I'll need to do something about this to get auto-formatting working on things like `let` and `do` blocks, but I don't think I need that behaviour quite yet, so I'll remove it for now and re-visit later 👍🏼 .